src/mlpack/core/cereal/array_wrapper.hpp

```diff
@@ -0,0 +1,86 @@
+/**
+ * @file core/cereal/array_wrapper.hpp
+ * @author Omar Shrit
+ *
+ * Implementation of an array wrapper.
+ *
+ * This implementation will allows to serilize array easily using cereal.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CEREAL_ARRAY_WRAPPER_HPP
+#define MLPACK_CORE_CEREAL_ARRAY_WRAPPER_HPP
+
+#include <cereal/archives/binary.hpp>
+#include <cereal/archives/portable_binary.hpp>
+#include <cereal/archives/xml.hpp>
+#include <cereal/archives/json.hpp>
+
+namespace cereal {
+
+template<class T>
+class ArrayWrapper
+{
+/* This file add make_array functionality to cereal
+ * This functionality exist only in boost::serialization.
+ * Most part of this code are copied from ArrayWrapper in boost::serialization
+ */
+public:
+
+  ArrayWrapper(T* t, std::size_t s) :
+      ArrayAddress(t),
+      ArraySize(s)
+  {}
+
+  /* default implementation
+  * Cereal does not require to split member, it can do that internally
+  * If this is the case we can not implement optimized version, since
+  * the only possible one is optimized.
+  * Some verification needed...
+  * default implementation
+  */
+  template<class Archive>
+  void serialize(Archive &ar, const unsigned int /*version*/)
+  {
+   // default implemention does the loop
+    size_t c = count();
```

Comment:
If we are loading though, don't we have to deserialize the count too, then allocate the space for the array?